core/exchange_test.go

```diff
@@ -156,10 +165,12 @@ func TestExchange_StoreHistoricIfArchival(t *testing.T) {
 		require.NoError(t, err)
 		assert.True(t, has)
 
-		// ensure .q4 file was not stored
+		// ensure .q4 file was not stored if not IsEmptyEDS
+		// TODO(chatton): verify if this is the correct behavior. Does the added WaitForHeight
+		// make it so there are some headers that are not empty and so a different code path is followed?
 		has, err = store.HasQ4ByHash(ctx, h.DAH.Hash())
 		require.NoError(t, err)
-		assert.False(t, has)
+		assert.Equal(t, has, !share.DataHash(h.DataHash).IsEmptyEDS())
```

Comment:
❗ I want to flag this ❗ 

I don't fully understand the expected behaviour based on test inputs. I don't want to make this change just so the test passes without making sure it is okay.

Reply:
Yes the `assert.False(t, has)` fails. It looks like we are hitting [this](https://github.com/celestiaorg/celestia-node/blob/69487249b1e276a4aba58e644e281401acb67924/core/eds.go#L80) codepath for some reason.

I'm wondering if it has something to do with the `cctx.WaitForHeight(30)` I added above. I initially added the wait for height as I encountered this error

```
fetching signed block at height 28 from core: rpc error: code = Unknown desc = nil block meta for height 28
```

from the `ce.GetRangeByHeight(ctx, genHeader, 30)` call.

Reply:
@renaynay  So digging into the code a little bit more, it looks like things pass if we just use a smaller number of in the `GetRangeByHeight`. I passed 5 and it works, it seems that if you pass in more (20+) we start entering [this codepath](https://github.com/celestiaorg/celestia-node/blob/87bc6c891949c80d1a68389af45c8bae87588597/store/store.go#L132) i.e. there is a datahash that is not empty.

I'm not sure how to debug this further, do you have any suggestions?